src/AccessControlHierarchyBase.php

```diff
@@ -171,51 +219,45 @@ public function configSubmit(array &$form, FormStateInterface $form_state) {
    * {@inheritdoc}
    */
   public function checkEntityAccess(EntityInterface $entity, $op, AccountInterface $account, WorkbenchAccessManagerInterface $manager) {
-    // Deny is our default response.
-    $return = AccessResult::forbidden();
-
     // Check that the content type is configured.
     // @TODO: Right now this only handles nodes.
-    $type = \Drupal::entityTypeManager()->getStorage('node_type')->load($entity->bundle());
+    /** @var \Drupal\node\NodeTypeInterface $type */
+    $type = $this->entityTypeManager->getStorage('node_type')->load($entity->bundle());
     $active = $type->getThirdPartySetting('workbench_access', 'workbench_access_status', 0);
 
     if (!$active) {
-      $return = AccessResult::neutral();
+      return AccessResult::neutral();
     }
 
     // Get the field data.
-    $scheme = $manager->getActiveScheme();
-    $field = $scheme->fields('node', $type->id());
+    $field = $this->fields('node', $type->id());
 
     // @TODO: Check for super-admin?
     // We don't care about the View operation right now.
     if ($op == 'view' || $account->hasPermission('bypass workbench access')) {
-      $return = AccessResult::neutral();
+      return AccessResult::neutral();
     }
-    elseif ($active && !empty($scheme) && !empty($field)) {
+    elseif ($active && !empty($field)) {
       // Discover the field and check status.
       $entity_sections = $this->getEntityValues($entity, $field);
       // If no value is set on the entity, ignore.
       // @TODO: Is this the correct logic? It is helpful for new installs.
       if (empty($entity_sections)) {
-        $return = AccessResult::neutral();
+        return AccessResult::neutral();
```

Comment:
early returns make this method much easier to follow the flow of, it then became clear that one of the returns here is the issue reported on d.o - content with empty sections is allowed regardless of wba being enabled for node type